src/browser/BrowserService.cpp

```diff
@@ -1009,12 +1009,27 @@ bool BrowserService::removeFirstDomain(QString& hostname)
 bool BrowserService::handleEntry(Entry* entry, const QString& url, const QString& submitUrl)
 {
     // Use this special scheme to find entries by UUID
-    if (url.startsWith("keepassxc://")) {
+    if (url.startsWith("keepassxc://by-uuid/")) {
         return ("keepassxc://by-uuid/" + entry->uuidToHex()) == url;
+    } else if (url.startsWith("keepassxc://by-path/")) {
+        return ("keepassxc://by-path/" + getPath(entry)) == url;
     }
     return handleURL(entry->url(), url, submitUrl);
 }
 
+QString BrowserService::getPath(Entry* entry) const
```

Comment:
I'm really not sure, but could this method be added as method of `Entry` instead? It feels like it would be better to put it there.